eval_test.go

```diff
@@ -389,23 +397,87 @@ func TestEvalPointerExpressions(t *testing.T) {
 
 	var i *int
 	data.Set("intPointer", &i)
-	RunJetTest(t, data, nil, "IntPointer_i", `{{ intPointer }}`, "")
+	RunJetTest(t, data, nil, "IntPointer_i", `{{ intPointer }}`, "<nil>")
+
+	st := StringerType{}
+	data.Set("stringerType", &st) // stringerType won't be dereferenced
+	RunJetTest(t, data, nil, "StringerTypePointer", `{{ stringerType }}`, "StringerType implements fmt.Stringer")
+
+	st2 := &StringerType{}
+	data.Set("stringerType2", &st2) // stringerType2 will be dereferenced just once
+	RunJetTest(t, data, nil, "StringerTypePointer2", `{{ stringerType2 }}`, "StringerType implements fmt.Stringer")
+
+	u := User{
+		Name:  "Pablo Escobbar",
+		Email: "pablo.escobar@cartel.mx",
+	}
+	data.Set("user", &u) // user will be dereferenced once
+	RunJetTest(t, data, nil, "UserPointer", `{{ user }}`, "{Pablo Escobbar pablo.escobar@cartel.mx}")
+
+	u1 := &u
+	data.Set("user1", &u1) // user1 will be dereferenced twice
+	RunJetTest(t, data, nil, "UserPointer1", `{{ user1 }}`, "{Pablo Escobbar pablo.escobar@cartel.mx}")
+
+	u2 := &u1
+	data.Set("user2", &u2) // user2 will be dereferenced only twice
+	RunJetTest(t, data, nil, "UserPointer2", `{{ user2 }}`, "&{Pablo Escobbar pablo.escobar@cartel.mx}")
 }
 
 func TestEvalPointerLimitNumberOfDereferences(t *testing.T) {
 	var data = make(VarMap)
 
 	var i *int
 	data.Set("intPointer", &i)
-	RunJetTest(t, data, nil, "IntPointer_i", `{{ intPointer }}`, "")
+	RunJetTest(t, data, nil, "IntPointer_i", `{{ intPointer }}`, "<nil>")
 
 	j := &i
 	data.Set("intPointer", &j)
-	RunJetTest(t, data, nil, "IntPointer_j", `{{ intPointer }}`, "")
+	RunJetTest(t, data, nil, "IntPointer_j", `{{ intPointer }}`, "<nil>")
+
+	// k := &j
```

Comment:
Can you drop these? This case is already tested in the previous test.

Reply:
This last test are supposed to show that a `***int` won't be resolved all the way to `<nil>` and instead would print as a pointer address, but there's no way to predict what that address will be. I can take them out of course.

Reply:
Yeah, I'd drop the commented lines. After that, we can merge this!

Reply:
Ah, completely forgot about this: 
f30e9c7